### PR TITLE
Add component for finding duplicates

### PR DESCRIPTION
**Pull request by rkondratowicz**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/DuplicateFinder.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.azure.storage.blob.BlobServiceClient;
+import uk.gov.hmcts.reform.blobrouter.data.model.Envelope;
+import uk.gov.hmcts.reform.blobrouter.services.EnvelopeService;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+public class DuplicateFinder {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/DuplicateFinder.java]: `DuplicateEnvelopeFinder` may be ?

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/DuplicateFinder.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    public DuplicateFinder(
+        BlobServiceClient storageClient,
+        EnvelopeService envelopeService
+    ) {
+        this.storageClient = storageClient;
+        this.envelopeService = envelopeService;
+    }
+
+    public List<Envelope> findIn(String containerName) {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/DuplicateFinder.java]: may be worth making it more specific ? `findEnvelopesMarkedDeleted` or something similar ?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/DuplicateFinder.java]: it's finding duplicates

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/DuplicateFinder.java]: it'd finding duplicates in given container. Notice it iterates over blobs.

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/DuplicateFinder.java]: Changed return type to `BlobItem` for clarity (all we actually need is file 'identifier' (name + container), but `Envelope` might have been misleading)

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/DuplicateFinder.java]: (I may add a `Duplicate` model later if it turns out we need the envelope ID)

**Pull request merged**